From 24c78aa70a37393d95b94020bca6b8384a10b3e2 Mon Sep 17 00:00:00 2001
From: Jamal Clarke <jamal_clarke@hotmail.co.uk>
Date: Tue, 13 May 2025 23:33:49 +0100
Subject: [PATCH] Averaging fft data timing

- Not isolated as the FFT task is the highest priority one therefore
  cannot be interrupted
- Timing is for the averaging of the FFT bin magnitudes, concerned with
  the time it takes to go from low to high
- Must apply the initial gpio debug patch prior to this one
---
 src/rtos/fft_rtos.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/rtos/fft_rtos.c b/src/rtos/fft_rtos.c
index f2e93d8..5a30b12 100644
--- a/src/rtos/fft_rtos.c
+++ b/src/rtos/fft_rtos.c
@@ -2,6 +2,7 @@
 #include "extern_i2s_dma_data.h"
 #include "mic_data_processing.h"
 #include "fft_processing.h"
+#include "stm32g4xx_gpio_debug.h"
 
 #include "stm32g4xx_hal.h"
 #include "stm32g4xx_nucleo.h"
@@ -91,6 +92,7 @@ void fft_task_processing(void* pvParameters)
 		// ignore DC component, any gather real frequencies and Nyquist
 		arm_cmplx_mag_f32(&fft_input[2], &bin_mags[fft_counter][0], FFT_DATA_SIZE/2);
 
+		deassert_gpio_debug_pin();
 		fft_counter += 1;
 		if (fft_counter == (int) (FFT_AVERAGE * 2)) {
 			average_bin_2d_array(FFT_AVERAGE, FFT_DATA_SIZE/2, &bin_mags[FFT_AVERAGE]
@@ -105,5 +107,6 @@ void fft_task_processing(void* pvParameters)
 
 			xQueueSendToFront(xFftCompleteFlagQueue, &fft_counter, pdMS_TO_TICKS(2));
 		}
+		assert_gpio_debug_pin();
 	}
 }
-- 
2.49.0

